### ISSUE TITLE
getting warning to upgrade on newest nvim version

### ISSUE THREAD

**Issue by Walkability** [neoclide/coc.nvim#2565]
I'm getting this warning:
 `coc.nvim works best on neovim >= 0.4.3`

even though I'm using.. `NVIM v0.5.0-812-gd17e50879`

**Comment by fannheyward** [comment]
What's the output `echo has('nvim') && !has('nvim-0.4.3')`?

**Comment by fatso83** [comment]
@fannheyward : Getting this as well. I know the bug is closed, but I'll add my output here anyway, in case it's worth reopening:

Output of `echo has('nvim')` is 1 and `echo has('nvim-0.4.3')` is 1, so unsurprisingly the output of `echo has('nvim') && !has('nvim-0.4.3')` is 0.

```
$ git rebase -i HEAD~10
Created autostash: 62e9cd6
hint: Waiting for your editor to close the file... coc.nvim works best on vim >= 8.1.1719 and neovim >= 0.4.0, consider upgrade your vim.
You can add this to your vimrc to avoid this message:
    let g:coc_disable_startup_warning = 1
Note that some features may behave incorrectly.
```

```
$ nvim --version
NVIM v0.4.4
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -g -O2 -fdebug-prefix-map=/build/neovim-Fc1Pxf/neovim-0.4.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=1 -DDISABLE_LOG -Wdate-time -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim-Fc1Pxf/neovim-0.4.4/build/config -I/build/neovim-Fc1Pxf/neovim-0.4.4/src -I/usr/include -I/usr/include/lua5.1 -I/build/neovim-Fc1Pxf/neovim-0.4.4/build/src/nvim/auto -I/build/neovim-Fc1Pxf/neovim-0.4.4/build/include
Compiled by team+vim@tracker.debian.org

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Comment by fatso83** [comment]
@chemzqm I see this bug was closed a day after reporting, with no explanation. Is there anything we could do to increase the likelihood of this being reopened?

**Comment by fatso83** [comment]
I found the bug! This check is wrong:
```
  if !has('nvim-0.4.0') && !has('patch-8.1.1719')
```

I can supply a PR

**Comment by chemzqm** [comment]
> I found the bug! This check is wrong:
> 
> ```
>   if !has('nvim-0.4.0') && !has('patch-8.1.1719')
> ```
> 
> I can supply a PR

How could it wrong? `has('nvim-0.4.0')` returns 0 for your neovim 0.4.4?

**Comment by fatso83** [comment]
LOL, this was an interesting exercise. I ended up spending the better part of the day writing a POC test framework for vim to enable writing a regression test for stubbing out the built-ins so that I could control what `has(value)` would return when given different values :neckbeard:  And of course, once that was in place, [the regression test](https://github.com/fatso83/coc.nvim/commit/6d7d0d3461dad2bc03a757ad4654e94d07b0f97a#diff-515ed18ea8a059f460935c8eed53566e44b92094a6798d84d0e8d751319a7f9dR11) showed that the code worked fine 🤡 

And indeed you are right, @chemzqm , there was no error in that code (which I now even proved to myself after learning VimScript🕵️). I was just confused by going back and forth too many times.

But I was curious: now that it turned out that the test passed once I controlled all the variables, why did I still get the warning? I mean, it turned up when rebasing my commits while doing this. So I ventured into finding out if I could see which version I was running when I saw these errors, as I suspected I was not running NeoVim after all. That introduced me to `:version`, which lo and behold showed that I was _really_ running Vim version 8.0.1453 ... Why? Because `git config core.editor` was set to `vim`, which overrode my shell's `EDITOR` setting. 

I suspect @Walkability was struck by something similar.
